src/parsers/js/transformers/eslint/index.js

```diff
@@ -0,0 +1,66 @@
+import compileModule from '../../../utils/compileModule';
+import pkg from 'eslint/package.json';
+
+const ID = 'eslint1';
+
+export default {
+  id: ID,
+  displayName: ID,
+  version: pkg.version,
+  homepage: pkg.homepage,
+
+  defaultParserID: 'acorn-to-esprima',
+
+  loadTransformer(callback) {
+    require(
+      [
+        // Explicitly require just the stuff we care about to avoid loading
+        // RuleTester and CLIEngine, which are unnecessary and bloat out the
+        // package size.
+        'eslint/lib/eslint',
+        'eslint/lib/util/source-code',
+        'eslint/lib/rules',
+        'babel-core',
+        'babel-eslint'
+      ],
+      (eslint, eslintSourceCode, eslintRules, babel, babelESLint) => {
+        callback({ eslint, eslintSourceCode, eslintRules, babel, babelESLint });
+      }
+    );
+  },
+
+  transform({ eslint, eslintSourceCode, eslintRules, babel, babelESLint }, transformCode, code) {
+    // Compile the transform code and install it as an ESLint rule. The rule
+    // name doesn't really matter here, so we'll just use a hard-coded name.
+    const rule = compileModule(
+      babel.transform(transformCode).code
+    );
+    eslintRules.define('astExplorerRule', rule);
+
+    // Run the ESLint rule on the AST of the provided code.
+    // Reference: http://eslint.org/docs/developer-guide/nodejs-api
+    const ast = babelESLint.parseNoPatch(code);
+    const sourceCode = new eslintSourceCode(code, ast);
+    const results = eslint.verify(sourceCode, {
+      rules: {
+        astExplorerRule: 2
+      }
+    });
+    return formatResults(results);
+  },
+};
+
+function formatResults(results) {
+  return results.length === 0
+    ? 'Lint rule not fired.'
+    : results.map(formatResult).join('').trim();
+}
+
+function formatResult(result) {
+  var pointer = '-'.repeat(result.column - 1) + '^';
+  return `
+\u2022 ${result.message} (at ${result.line}:${result.column})
```

Comment:
@RReverser  - Because the code could be a partial snippet that might break the syntax highlighting. Consider the trivial "Do not use template literals" rule I included as an example. If you have a multi-line template literal, the syntax highlighting would break due to the fact that the template literal is not closed:

<img width="1253" alt="screen shot 2016-03-22 at 8 11 25 pm" src="https://cloud.githubusercontent.com/assets/91933/13974603/5e16a684-f06a-11e5-81e1-0113efc09317.png">

(see the orange carat)

I don't think it's safe to syntax highlight partial code snippets unless they're all individually highlighted.